### ISSUE TITLE
Errors during upgrade process from 2.1.27.36 to 3.0.0.14

### ISSUE THREAD

**Issue by ewheelerinc** [nilsteampassnet/TeamPass#3086]
### Steps to reproduce
1. Install fresh 3.0.0.14
2. Go through /install/install.php process and reference the correct (2.1.27.36) salt key during installation
3. Import database from 2.1.27.36 over the top of the 3.0.0.14 install
4. Try to go through the /install/upgrade.php process

### Expected behavior
Upgrade process succeeds

### Actual behavior
Upgrade process fails to connect to the database but only gives cryptic errors in the browser console

### Server configuration
**Operating system**: Oracle Linux 8

**Web server:** Apache 2.4.37

**Database:** MariaDB 

**PHP version:** 10.3.28

**Teampass version:** 3.0.0.14

**Teampass configuration file:**

[tp.config.php.gz](https://github.com/nilsteampassnet/TeamPass/files/8257934/tp.config.php.gz)

**Updated from an older Teampass or fresh install:**

upgrade 2.1.27.36 to 3.0.0.14

### Client configuration
**Browser:** Firefox 91.5.0esr

**Operating system:** Linux

#### Web server error log

No log output was generated in this process

#### Log from the web-browser developer console (CTRL + SHIFT + i)
```
Uncaught SyntaxError: JSON.parse: expected ',' or '}' after property value in object at line 2 column 71 of the JSON data
    complete https://teampass.ewi/install/upgrade.php:216
    jQuery 6
    <anonymous> https://teampass.ewi/install/upgrade.php:208
    jQuery 9
    <anonymous> https://teampass.ewi/install/upgrade.php:151
    jQuery 13

```

Response:

```
[{ "error" : "Impossible to get connected to server. Error is: " . addslashes(mysqli_connect_error()), "index" : "" }]
```

#### Recommendation
It sounds like we have a database connection problem. It would be nice to get a friendly error indicating that since not everyone knows how to debug web applications.

**Comment by nilsteampassnet** [comment]
WHy do you proceed as this?
Upgrade process should not start with an install of the latest.
You should upgrade the existing one.
Do you backups (DN and files)
Download and extract the new files in new folder
Copy files tp.config.php, settings.php, csrfp.config.php
Copy folders files and upload
Run upgrade

**Comment by ewheelerinc** [comment]
Thank you for providing the upgrade procedure, that worked for us.

That being said, we can no longer modify any of the entries. When we try, we get the error "Expected level of security not reached!". I have verified that the roles are correct (and in fact the user I am testing on is a manager, and under options, " Managers may edit or delete any items they can view" is checked).

I also tried the following:
- Re-apply the permissions for the role in question.
- Create a new user with the correct roles and gave them admin privileges

What else can we try? Is there any debugging we can turn on to track down this issue?
